packages/components/src/TabContainer.tsx

```diff
@@ -0,0 +1,65 @@
+import { ActionButton, ActionButtonProps } from "ActionButton";
+import clsx from "clsx";
+import { useState } from "react";
+import { twMerge } from "tailwind-merge";
+
+type TabContainerProps = {
+  id: string;
+  title: string;
+  tabs: Array<{
+    title: React.ReactNode;
+    children: React.ReactNode;
+  }>;
+} & ActionButtonProps<"button">;
+
+export const TabContainer = ({
+  id,
+  title,
+  tabs,
+  ...buttonProps
+}: TabContainerProps): JSX.Element => {
+  const [activeTabIndex, setActiveTabIndex] = useState(0);
+
+  return (
+    <div>
+      <div role="tablist" aria-label={title} className="flex">
+        {tabs.map((tab, index) => (
+          <ActionButton
+            key={index}
+            id={`tab-${id}-${index}`}
+            role="tab"
+            aria-selected={activeTabIndex === index}
+            aria-controls={`tabpanel-${id}-${index}`}
+            tabIndex={activeTabIndex === index ? 0 : -1} // Only the active tab is focusable
```

Comment:
personally, I prefer to unset the tabIndex directly on the component because it can easily conflict with other input elements on the page and mess it around, then usually I define it on the page level. But we can test it as well and observe if it's necessary or not :)